### DISCUSSION TITLE
Two limits on each axis for STM32F446ZE with other interrupt inputs

### DISCUSSION THREAD

**Discussion by samitray** [grblHAL/core#528] [General]
Trying to make a map file for STM32F446ZE custom board, since there are lots of pins, we can do 2 pins for each axis, which technically means that it is possible to have 12 pins connected to limits (2 limits on ABC make sense only in certain scenarios). That leaves exactly 4 pins with interrupt capabilities (out of 16), which are for RST, HOLD, START and PROBE then we are out of interrupt capable pins.
Is there any other way to try add few more interrupt for DOOR and alternate PROBE. I could drop 2 of the limits (easy but not ideal) or add IO expander chip on one interrupt line and query on interrupt (plugin - difficult given my programming skills). Or put both limits of each axis on the same pin but different port (say PB9 and PC9) assuming that any same pin on different port will cause the interrupt to fire, this is similar to NC switches in series, but potentially plugin code could check pin status and determine which limit was hit. No changes in core or homing is a prerequisite. Let me check if two input on same channel on different port trigger the same external interrupt

**Comment by terjeio** [comment]
> Let me check if two input on same channel on different port trigger the same external interrupt

AFAIK no it does not.

However you _can_ [configure Z min for polling](https://github.com/grblHAL/STM32F4xx/blob/1cfbda75bd71d59eb73aa3c700baf5d526722e1a/Src/driver.c#L3342-L3353) too free up one pin.

Using an IO expander is possible via board specific code. But then the homing end should still be routed directly to avoid latency issues?

**Reply by samitray** [reply]
> Using an IO expander is possible via board specific code. But then the homing end should still be routed directly to avoid latency 

> issues?

Yes the homing/limit pins will have to go directly to interrupt, extra limits through IO expander and single interrupt from io expander into interrupt pin with plugin....
Also series NC switches would work, if there is no need to know which limit was triggered, 
for now i need the motors to stop if any limit is triggered, so series NC limit switched should work...
Thanks
